### ISSUE TITLE
collapsible category

### ISSUE THREAD

**Issue by user00843** [eckmarcommunity/eckmar#44]
categories should be collapsible in case of a lot of subcategories to reduce page clutter, there is partial written accordion rule in css file but is not integrated in categories.blade.
I think its possible to do it with html label
`       input[type=checkbox] ~.subcategories{
             display: none;
       }
        input[type=checkbox]:checked ~.subcategories{
              display: block;
               margin-bottom: 3px !important;
    }
       .category input[type="checkbox"] {
 display: none;
 position: relative;`

can you implement this the correct way?

**Comment by dclipca** [comment]
> categories should be collapsible in case of a lot of subcategories to reduce page clutter, there is partial written accordion rule in css file but is not integrated in categories.blade. I think its possible to do it with html label ` input[type=checkbox] ~.subcategories{ display: none; } input[type=checkbox]:checked ~.subcategories{ display: block; margin-bottom: 3px !important; } .category input[type="checkbox"] { display: none; position: relative;`
> 
> can you implement this the correct way?

I would look into implementing this feature when I have some free time.

**Comment by MikeOxlong6713** [comment]
> categories should be collapsible in case of a lot of subcategories to reduce page clutter, there is partial written accordion rule in css file but is not integrated in categories.blade. I think its possible to do it with html label ` input[type=checkbox] ~.subcategories{ display: none; } input[type=checkbox]:checked ~.subcategories{ display: block; margin-bottom: 3px !important; } .category input[type="checkbox"] { display: none; position: relative;`
> 
> can you implement this the correct way?

Do you still need help with this? I managed to be able to do it.

**Comment by MikeOxlong6713** [comment]
This is how I did mine but I also have a subcat for Samples.

eckmar/resources/views/includes

categories.blade.php

`<style>
    .nav-group.ticker-menu-group {
        margin-top: .25rem;
        font-size: small;
    }

    .nav-group .nav-list.ticker-menu {
        padding: 0 .5rem 0 .5rem;
    }

    .nav-group .nav-list .ticker-currency {
        font-weight: bold;
    }

    .nav-group .nav-list .ticker-change {
        float: right;
    }

    .nav-group .nav-list .ticker-price {
        float: right;
    }

    .ticker-change {
        font-size: x-small;
    }

    .ticker-change.trend-positive {
        color: #6fb956;
    }

    .ticker-change.trend-positive:before {
        color: #6fb956;
        content: '+';
    }

    .ticker-change.trend-negative {
        color: #f13232;
    }

    .ticker-change.trend-negative:before {
        color: #f13232;
    }

    .table-form .table {
        margin-top: 0;
    }

    .table th a {
        color: #565656;
    }

    .form-wrapper {
        max-width: 600px;
    }

    .form-group-wide {
        padding-left: .5rem !important;
        margin-bottom: 0 !important;
    }

    .table .column-small {
        width: 35px;
    }

    .table td.right,
    .table th.right {
        text-align: right;
    }

    .table.table-no-margin {
        margin-top: 0;
    }

    .pull-right {
        float: right;
    }

    .pull-left {
        float: left;
    }

    .clr-form.clr-form-compact .clr-form-control {
        margin-top: 0;
    }

    .nav-list.ticker-menu {
        list-style-type: none;
        padding: 0;
    }

    .nav-list.ticker-menu li {
        margin-bottom: 0.5rem;
    }
</style>

<h4>Samples</h4>
<div class="list-group categories">
    @if ($samples = $categories->firstWhere('name', 'Samples'))
        @include('includes.subcategories', ['categories' => $samples->children])
    @endif
</div>

<h4>Categories</h4>
<div class="list-group categories">
    @include('includes.subcategories', ['categories' => $categories->reject(function ($category) {
        return $category->name === 'Samples';
    })])
</div>

<section class="sidenav-content" style="background-color: #46586a !important;padding: 10px;color: whitesmoke; text-align: start;">
    <div id="crypto-prices" class="crypto-prices"></div>
    <hr>

    <div class="app-version">
        <div>
            <small>© 2023 DRUGBAY Market</small>
        </div>
        <div>
            <small>Version: 1.2.9.16</small>
        </div>
    </div>
</section>


<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>`

subcategories.blade.php

`@foreach($categories as $cat)
    <div class="category">
        <div class="list-group-item category @if(isset($category) && $cat -> isAncestorOf(optional($category))) active @endif  align-items-center rounded-0 list-group-item-action d-flex justify-content-between">
            @if($cat -> children -> isNotEmpty())
                <a href="#collapse-{{$cat->id}}" class="d-inline-block" data-toggle="collapse">
                    <span class="expand-collapse-indicator">+</span>
                </a>
            @endif
            <a href="{{ route('category.show', $cat) }}">
                {{ $cat -> name }}
            </a>
            <span class="badge badge-warning badge-pill">{{ $cat -> num_products }}</span>
        </div>
        @if($cat -> children -> isNotEmpty())
            <div class="pl-3 subcategories collapse" id="collapse-{{$cat->id}}">
                @include('includes.subcategories', ['categories' => $cat -> children])
            </div>
        @endif
    </div>
@endforeach`

hope this helps someone else. If you don't want the samples remove it as it adds Samples to a different Category with the subcategories for this.

**Comment by MikeOxlong6713** [comment]
Here is how I did it, you might want to remove the option for Samples if that's not what you want.

categories.blade.php

```
<h4>Categories</h4>
<hr>
<div class="list-group categories">
    @include('includes.subcategories', ['categories' => $categories])
</div>
```

includes.subcategories.blade.php

```
@foreach($categories as $cat)
    <div class="category">
        <a data-toggle="collapse" href="#category-{{$cat->id}}" role="button" aria-expanded="false" aria-controls="category-{{$cat->id}}" class="list-group-item category @if(isset($category) && $cat -> isAncestorOf(optional($category))) active @endif  align-items-center rounded-0 list-group-item-action d-flex justify-content-between">
            {{ $cat -> name }}
            <span class="badge badge-warning badge-pill">{{ $cat -> num_products }}</span>
        </a>
        @if($cat -> children -> isNotEmpty())
            <div class="pl-3 subcategories collapse" id="category-{{$cat->id}}">
                @include('includes.subcategories', ['categories' => $cat -> children])
            </div>
        @endif
    </div>
@endforeach
```

master.main.blade.php

```
<!doctype html>
<html lang="en">
<head>
    <!-- Required meta tags -->
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

    <!-- Bootstrap CSS -->
    <link rel="stylesheet" href="/css/app.css">

    @hasSection('title')
        <title>{{config('app.name')}} - @yield('title')</title>
    @else
        <title>{{config('app.name')}}</title>
    @endif
</head>
<body class="pb-4">
    @include('master.navbar')
    @include('master.search')

    @hasSection('container-fluid')
        <div class="container-fluid">
    @else
        <div class="container">
    @endif
            @include('includes.jswarning')
            <div class="mt-4">
                @yield('content')
            </div>
        </div>

    <!-- jQuery script -->
    <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js"></script>

    <!-- Bootstrap script -->
    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.min.js"></script>

</body>
</html>
```
